quam/components/channels.py

```diff
@@ -1491,3 +1452,49 @@ class InIQOutSingleChannel(SingleChannel, InIQChannel):
     """
 
     pass
+
+
+def _config_add_opx_controller(config: Dict[str, dict], controller_name: str) -> Dict[str, dict]:
+    """Adds a controller to the config if it doesn't exist, and returns its config.
+
+    config.controllers.<controller_name> will be created if it doesn't exist.
+    It will also add the analog_outputs, digital_outputs, and analog_inputs keys
+
+    Args:
+        config (dict): The QUA config that's in the process of being generated.
+        controller_name (str): The name of the controller.
+
+    Returns:
+        Dict[str, dict]: The config entry for the controller.
+    """
+    config["controllers"].setdefault(controller_name, {})
+    controller_cfg = config["controllers"][controller_name]
+    for key in ["analog_outputs", "digital_outputs", "analog_inputs"]:
+        controller_cfg.setdefault(key, {})
+
+    return controller_cfg
+
+
+def _config_add_opx1000_controller(config: Dict[str, dict], controller_name: str, fem_idx: int) -> Dict[str, dict]:
+    """Adds a controller to the config if it doesn't exist, and returns its config.
+
+    config.controllers.<controller_name> will be created if it doesn't exist.
+    It will also add the analog_outputs, digital_outputs, and analog_inputs keys
+
+    Args:
+        config (dict): The QUA config that's in the process of being generated.
+        controller_name (str): The name of the controller.
+
+    Returns:
+        Dict[str, dict]: The config entry for the controller.
+    """
+    config["controllers"].setdefault(controller_name, {})
+    controller_cfg = config["controllers"][controller_name]
+    controller_cfg.setdefault("fems", {})
+    fem_config = controller_cfg["fems"]
+    fem_config.setdefault(fem_idx, {"type": "LF"})
+    fem_config = fem_config[fem_idx]
```

Comment:
`setdefault` returns the element, so these lines should be simplified: `fem_config = controller_cfg.setdefault("fems", {})`

Reply:
Fixed